llvm/include/llvm/CodeGen/GlobalISel/LoadStoreOpt.h

```diff
@@ -35,11 +35,29 @@ struct LegalityQuery;
 class MachineRegisterInfo;
 namespace GISelAddressing {
 /// Helper struct to store a base, index and offset that forms an address
-struct BaseIndexOffset {
+class BaseIndexOffset {
+private:
   Register BaseReg;
   Register IndexReg;
-  int64_t Offset = 0;
-  bool IsIndexSignExt = false;
+  std::optional<int64_t> Offset;
+
+public:
+  BaseIndexOffset() = default;
+  BaseIndexOffset(Register Base, Register Index, bool IsIndexSignExt)
```

Comment:
`IsIndexSignExt` dead?

Reply:
Yes our implementation doesn't currently use that so I removed it.

Reply:
The constructor is unused and there is no member variable.